test/server.test.js

```diff
@@ -2957,3 +2957,22 @@ test('calling next twice should throw', function (t) {
         t.ifError(err);
     });
 });
+
+test('aborting pre should still call after', function (t) {
```

Comment:
Can we put this at/around L2312 to group it with the other pre handler test that is testing `next(false)`? And maybe rename the test to `next(err) should still trigger after event` to be a little more precise since "abort" doesn't really mean much in context of existing restify docs or nomenclature.